internal/pkg/gateway/api.go

```diff
@@ -273,6 +270,43 @@ func (gs *Server) Endorse(ctx context.Context, request *gp.EndorseRequest) (*gp.
 	return endorseResponse, nil
 }
 
+// responseStatus unpacks the proposal response and error values that are returned from ProcessProposal and
+// determines how the gateway should react (retry?, close connection?).
+// Uses the grpc canonical status error codes and their recommended actions.
+// Returns:
+// - response successful (bool)
```

Comment:
Good that you mention this is (bool). I'd recommend do the same for all four parameters so that reader can keep track of them.